diff-hl-show-hunk.el

```diff
@@ -0,0 +1,326 @@
+;;; diff-hl-show-hunk.el --- Integrate popup/posframe and diff-hl-diff-goto-hunk -*- lexical-binding: t -*-
+
+;; Copyright (C) 2020  Free Software Foundation, Inc.
+
+;; Author: Álvaro González <alvarogonzalezsotillo@gmail.com>
+
+;; This file is part of GNU Emacs.
+
+;; GNU Emacs is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; GNU Emacs is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; `diff-hl-show-hunk' shows a posframe/popup with the modified hunk at point.
+;; `diff-hl-show-hunk-function' contains the backend used to show the hunk.  Its
+;; default value is `diff-hl-show-hunk-function-default', which tries to use
+;; `diff-hl-show-hunk-posframe' (GUI), and then `diff-hl-show-hunk-popup'.
+;; Other backends (for example pos-tip or phantom overlays) could be
+;; implemented.
+;;
+;; It uses the same diff algorithm as `diff-hl-flydiff-mode', so it is
+;; highly recomended to enable `diff-hl-flydiff-mode' before use this package.
+;;
+;; `diff-hl-show-hunk-mode' shows the posframe/popup when clicking
+;; in the margin or the fringe.
+;;
+;; To use it in all buffers:
+;;
+;;    ```
+;;      (global-diff-hl-show-hunk-mode)
+;;    ```
+;;
+;; Two hunks can overlap due to the contex of the diff algorithm.  To
+;; avoid this overlap, set 0 contex lines in `vc-diff'.  The following
+;; backends are automatically configured:
+;;
+;; - GIT: `vc-git-diff-switches' uses '--unified=0'.
```

Comment:
You document these switches here, but they are actually reused from the existing code in `diff-hl`, right?

Reply:
You are right, this is an outdated comment of a previous commit (before implementing the `diff-hl-diff-buffer-with-head` thing).
I am removing it.